### PR TITLE
Feat/restructuring

### PR DESCRIPTION
**Pull request by jkiddo**
This PR moves a lot around on the classes and puts the construction of the FHIR server into a separate bean for making it easier for other people to reuse it. All field injection has also been removed and put into constructors or bean creatino instead. There is still much more work to be done as with this redesign as eg. all the inceptors can be retrofitted later in construction process. This step however is the first in getting a more cleaner setup and the first step for us to be able to do more auto configuration more cleanly. Let me know what you think @tadgh @michaelabuckley @patrick-werner

### CODE REVIEW

**Comment by jkiddo** [comment]
Ideally all the optional arguments can also be removed over time but before I go into such endeavours I would like to hear what you think of it so far.

File: src/main/java/ca/uhn/fhir/jpa/starter/Application.java [hunk 1]
```diff
@@ -56,11 +57,10 @@ protected SpringApplicationBuilder configure(
...
   @Bean
   @Conditional(OnEitherVersion.class)
-  public ServletRegistrationBean hapiServletRegistration() {
+  public ServletRegistrationBean hapiServletRegistration(RestfulServer restfulServer) {
     ServletRegistrationBean servletRegistrationBean = new ServletRegistrationBean();
-    JpaRestfulServer jpaRestfulServer = new JpaRestfulServer();
-    beanFactory.autowireBean(jpaRestfulServer);
-    servletRegistrationBean.setServlet(jpaRestfulServer);
+    beanFactory.autowireBean(restfulServer);
+    servletRegistrationBean.setServlet(restfulServer);
```
Comment (tadgh) [inline src/main/java/ca/uhn/fhir/jpa/starter/Application.java:63]: Nice, I like this refactor already

**Review by tadgh — APPROVED** [review]
Approved pending one question

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ hapi:
     #    default_pretty_print: true
     #    default_page_size: 20
     #    delete_expunge_enabled: true
-    #    enable_repository_validating_interceptor: false
+    enable_repository_validating_interceptor: true
```
Comment (tadgh) [inline src/main/resources/application.yaml:92]: Question (non-blocking): Did you intend to set this to true during a refactor? Seems out of place

**Review by tadgh — APPROVED** [review]

**Review by oliveregger — APPROVED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (patrick-werner) [reply src/main/resources/application.yaml:92]: agree, this should be false by default

**Review by patrick-werner — APPROVED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (jkiddo) [reply src/main/resources/application.yaml]: That was a bug - only enabled for testing

**Pull request merged**